engines/sci/engine/klists.cpp

```diff
@@ -824,6 +827,13 @@ reg_t kArraySetElements(EngineState *s, int argc, reg_t *argv) {
 }
 
 reg_t kArrayFree(EngineState *s, int argc, reg_t *argv) {
+	if (s->_segMan->getSegmentType(argv[0].getSegment()) == SEG_TYPE_SCRIPT) {
```

Comment:
I was referring to the SCI3 deletion function. I stand corrected regarding the arrays segment. you rewrote that code in 3f91726765 and now its behavior is what you mention.